pkg/vsphrcoll/collector.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+//nolint:stylecheck // names should be as in rest output
+package vsphrcoll
+
+import "encoding/json"
+
+const (
+	virtualMachineQuery = "api/vcenter/vm"
+)
+
+type portResource struct {
+	Label                        string
+	Type                         string
+	Upt_compatibility_enabled    bool
+	Upt_v2_compatibility_enabled bool
+	Mac_type                     string
+	Mac_address                  string
+	Pci_slot_number              int
+	Wake_on_lan_enabled          bool
+	Backing                      struct {
+		Type                    string
+		Network                 string
+		Network_name            string
+		Host_device             string
+		Distributed_switch_uuid string
+		Distributed_port        string
+		Connection_cookie       int
+		Opaque_network_type     string
+		Opaque_network_id       string
+	}
+	State               string
+	Start_connected     bool
+	Allow_guest_control bool
+}
```

Comment:
can API types be imported similar to nsx?

Reply:
> can API types be imported similar to nsx?
I didnt dive into it. if yes, it will involve generation scrips as ziv wrote.
I left it for now

Reply:
maybe it's already available if `govmomi` ?
see for example:
https://github.com/vmware/govmomi/blob/main/vim25/types/json_test.go
https://github.com/vmware/govmomi/blob/main/vim25/types/testdata/vminfo.json